presto-main/src/main/java/com/facebook/presto/sql/planner/CompilerConfig.java

```diff
@@ -37,4 +40,30 @@ public CompilerConfig setExpressionCacheSize(int expressionCacheSize)
         this.expressionCacheSize = expressionCacheSize;
         return this;
     }
+
+    public int getLeafNodeLimit()
+    {
+        return this.leafNodeLimit;
+    }
+
+    @Config("planner.max-leaf-nodes-in-plan")
+    @ConfigDescription("Maximum number of leaf nodes in logical plan, throw an exception when exceed if leaf-node-limit-enabled is set true")
+    public CompilerConfig setLeafNodeLimit(int num)
+    {
+        this.leafNodeLimit = num;
+        return this;
+    }
+
+    public boolean getLeafNodeLimitEnabled()
+    {
+        return this.leafNodeLimitEnabled;
+    }
+
+    @Config("planner.leaf-node-limit-enabled")
+    @ConfigDescription("Throw an exception if number of leaf nodes in logical plan exceeds threshold set in max_leaf_nodes_in_plan")
```

Comment:
nit, max_leaf_nodes_in_plan -> max-leaf-nodes-in-plan

Reply:
Done.